modules/runner/src/test/java/org/apache/ignite/internal/configuration/compatibility/framework/ConfigurationTreeComparator.java

```diff
@@ -105,6 +114,22 @@ private ConfigNode find(ConfigNode node, Collection<ConfigNode> candidates) {
                 }
             }
 
+            if (deletedItems == null) {
+                deletedItems = new ArrayList<>(configModules.size());
+
+                for (ConfigurationModule module : configModules) {
+                    KeyIgnorer keyIgnorer = KeyIgnorer.fromDeletedPrefixes(module.deletedPrefixes());
+
+                    deletedItems.add(keyIgnorer);
+                }
+            }
+
+            boolean deleted = deletedItems.stream().anyMatch(i -> i.shouldIgnore(node.path()));
```

Comment:
Should we check deleted prefixes before trying to find a match?
I'd prefer to avoid using `ConfigurationModule` classes in comparator. Can we incapsulate prefixes and maybe the logic into ComparisonContext? 

`LeafNodesVisitor shuttle = new LeafNodesVisitor(new Validator(actualTrees, context));`
```
class Validator {
.....
private Validator(List<ConfigNode> roots, ComparisonContest context) {
....
private ConfigNode find(ConfigNode node, Collection<ConfigNode> candidates) {
   if (context.isDeleted(node)) {
       return ContextNode.DELETED_INSTANCE; /// or null ?
   }
   
   for (ConfigNode candidate : candidates) {
                if (match(node, candidate)) {
                .......
```

Reply:
done